src/Collection.php

```diff
@@ -520,9 +520,10 @@ public function find($filter = [], array $options = [])
             $options['typeMap'] = $this->typeMap;
         }
 
-        $operation = new Find($this->databaseName, $this->collectionName, $filter, $options);
+        //$operation = new Find($this->databaseName, $this->collectionName, $filter, $options);
+        //return $operation->execute($server);
 
-        return $operation->execute($server);
+        return new \MongoDB\Cursor($this, $filter, $options);
```

Comment:
This change conflicts with "It doesn't affect the original methods and usage", as we're changing the return type of the `find()` method. There is also a subtle behavioral change because the actual query's execution will now be deferred until iteration starts on the returned `MongoDB\Cursor` object. This could cause problems with server availability, since server selection for the query happens during the `find()` call (i.e. the code no longer immediately selects a server and executes a query in the same method call).